packages/astro/src/vite-plugin-scanner/scan.ts

```diff
@@ -51,33 +49,41 @@ export async function scan(
 
 	let pageOptions: PageOptions = {};
 	for (const _export of exports) {
-		const { n: name, le: endOfLocalName } = _export;
-		// mark that a `prerender` export was found
-		if (BOOLEAN_EXPORTS.has(name)) {
-			// For a given export, check the value of the local declaration
-			// Basically extract the `const` from the statement `export const prerender = true`
-			const prefix = code
-				.slice(0, endOfLocalName)
-				.split('export')
-				.pop()!
-				.trim()
-				.replace('prerender', '')
-				.trim();
-			// For a given export, check the value of the first non-whitespace token.
-			// Basically extract the `true` from the statement `export const prerender = true`
-			const suffix = code.slice(endOfLocalName).trim().replace(/\=/, '').trim().split(/[;\n]/)[0];
-			if (prefix !== 'const' || !(isTruthy(suffix) || isFalsy(suffix))) {
+		const { n: name, ls: startOfLocalName, le: endOfLocalName } = _export;
+		if (startOfLocalName === -1 || endOfLocalName === -1) continue;
+		// For a given export, check the value of the local declaration
+		// Basically extract the `const` from the statement `export const prerender = true`
+		const indexOfExport = code.lastIndexOf('export', startOfLocalName);
+		const prefix = code.slice(indexOfExport + 6, startOfLocalName).trim();
+
+		// `export const name = "value";
+		//                      ^      ^
+		// indexOfFirstNonWhitespace  endOfLineIndex
+		const indexOfFirstNonWhitespace = code.slice(endOfLocalName).search(/[^=\s]/) + endOfLocalName;
+		const endOfLineIndex = code.slice(indexOfFirstNonWhitespace).search(/[\n;]/) + indexOfFirstNonWhitespace;
```

Comment:
This check won't work for carriage returns, e.g. `\r`

Also, we can't assume there's a semicolon.

Reply:
- In the case of a carriage return, this will return `"value"\r` and the carriage return will be trimmed off on line 64
- It doesn't assume there's a semicolon – it splits off at either the first semicolon OR the first newline

Reply:
Added a test case for this and it passes